src/OrchardCore.Modules/OrchardCore.Users/Controllers/AccountController.cs

```diff
@@ -630,9 +633,9 @@ public async Task<IActionResult> LinkExternalLogin(LinkExternalLoginViewModel mo
 
                 return NotFound();
             }
-            var email = info.Principal.FindFirstValue(ClaimTypes.Email) ?? info.Principal.FindFirstValue("email");
-
-            var user = await _userManager.FindByEmailAsync(email);
+            //really important the order of the services registration in the dependency injection context
```

Comment:
No need for duplicating this comments. Add similar comment during the initialization. Even better is to lazy load the re-sorted service so it is not always sorted during the service construction.